frontend/src/components/Layout/Carousel/CarouselProgressBar.vue

```diff
@@ -24,6 +24,10 @@
 import { defineComponent } from 'vue';
 import { useResponsiveClasses } from '@/composables';
 
+interface Data {
```

Comment:
For this file, Swiper 9 includes a new autoplay module. Would you mind trying to move this component to use Swiper's autoplay API: https://swiperjs.com/swiper-api#autoplay (which after 9.X version allows for pausing, which is something that was not possible before)

Also, using swiper native paginators: https://swiperjs.com/swiper-api#pagination

This is something I was debating wether to do after or before vite branch gets merged because the earlier it gets merged, the merrier (so we don't have master frozen for longer), but now that you took the plunge to fix the linting errors here, do you want to take the challenge? :)

Reply:
I'm be super happy to pick this up as my first dive into non-typing changes now that I've had some time to poke around in the project and feel like I have a grasp of what's going. I'll do that as my next task, and I'll put it in a separate PR to compartmentalize the changes.